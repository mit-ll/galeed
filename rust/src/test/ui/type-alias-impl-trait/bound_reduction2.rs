#![feature(type_alias_impl_trait)]

fn main() {
}

trait TraitWithAssoc {
    type Assoc;
}

type Foo<V> = impl Trait<V>;
//~^ ERROR the trait bound `T: TraitWithAssoc` is not satisfied

trait Trait<U> {}

impl<W> Trait<W> for () {}

fn foo_desugared<T: TraitWithAssoc>(_: T) -> Foo<T::Assoc> {
    ()
}
