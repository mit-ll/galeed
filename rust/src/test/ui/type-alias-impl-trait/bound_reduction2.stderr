error[E0277]: the trait bound `T: TraitWithAssoc` is not satisfied
  --> $DIR/bound_reduction2.rs:10:15
   |
LL | type Foo<V> = impl Trait<V>;
   |               ^^^^^^^^^^^^^ the trait `TraitWithAssoc` is not implemented for `T`
   |
help: consider further restricting this bound
   |
LL | fn foo_desugared<T: TraitWithAssoc + TraitWithAssoc>(_: T) -> Foo<T::Assoc> {
   |                                    ^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
